### ISSUE TITLE
[docs] Metadata on CSTVisitor

### ISSUE THREAD

**Issue by thatch** [Instagram/LibCST#250]
1. We should make it more obvious in the examples how to use metadata with CSTVisitor; if you are making incremental improvements and already have a visitor, the docs don't make it super obvious that

```
module.visit(SomeClass())
```

becomes

```
wrapper = cst.MetadataWrapper(module)
module.visit(SomeClass())
```

2. If you forget that, `get_metadata()` just gives `KeyError`.  We can probably improve that, when there is no metadata to give a more descriptive error.

**Comment by jimmylai** [comment]
For 1. which docs aren't clear enough? 
We currently have a metadata API page and a metadata tutorial. Both of them present `MetadataWrapper` in the examples. What's your suggestion to make the docs more obvious?
https://libcst.readthedocs.io/en/latest/metadata.html#metadata-apis
https://libcst.readthedocs.io/en/latest/metadata_tutorial.html#Accessing-Metadata

For 2. I totally agree, the error is confusing to new users. If you get some time to fix it, PR is welcome!

**Comment by thatch** [comment]
I created a PR now.  Longer term, the merging typestub example is great, but I had to click around a lot to piece together how to use a visitor with a real world example, which contributed to my initial confusion.

I initially hit this while trying to write test cases for a visitor that has other methods that will use metadata.

```
class X(CSTVisitor):
  def visit_Name(self, ...):
    self.do_something_with(node.name)
  def do_something_with(self, name):
     self.get_metadata(ScopeProvider)[name]

class T(unittest.TestCase):
  def test_X_1(self):
    mod = ...
    x = X()
    MetadataWrapper(mod).visit(x)
    self.assertEqual(..., x.do_something_with("foo"))
```

It was not obvious from the docs that metadata is cleared after the visit is over.  We could probably use something a bit longer than the api example that exercises more  providers, or shows how to test it.  Maybe renaming a dotted name in the presence of `from` imports...
